lib/AST/Decl.cpp

```diff
@@ -3612,6 +3612,10 @@ void VarDecl::emitLetToVarNoteIfSimple(DeclContext *UseDC) const {
     // If the problematic decl is 'self', then we might be trying to mutate
     // a property in a non-mutating method.
     auto FD = dyn_cast_or_null<FuncDecl>(UseDC->getInnermostMethodContext());
+
+    // If this is a subscript getter, don't suggest adding mutating.
+    if (FD->getAccessorStorageDecl()->getGetter() == FD) return;
```

Comment:
Regarding fixits, @jrose-apple is the best one to answer that question.

In general, it's better to not emit a fixit than it is to crash, so I'm OK with taking the patch in its current form (once you move the test case). However if there's an easy way to generalize the fixit instead, that would be great.